src/cogent3/app/evo.py

```diff
@@ -104,29 +106,142 @@ def __init__(
+        >>> tree_func = dist_cal + est_tree
+        >>> model = get_app("model", "F81", tree_func=tree_func)
+        >>> result = model(aln2)
+        >>> print(result.to_rich_dict())
+        {'type': 'cogent3.app.result.model_result', ...
+
+        Specify the time-heterogeneous settings for the model fitting.
+        For details, see https://cogent3.org/doc/app/evo-model-timehet#
+
+        >>> app_thet = get_app(
+        ...    "model",
+        ...    "HKY85",
+        ...    tree=tree,
+        ...    time_het=[dict(tip_names=["Human", "Opossum"], outgroup_name="Mouse")],
+        ... )
+        >>> result = app_thet(aln2)
+        >>> print(result.to_rich_dict())
+        {'type': 'cogent3.app.result.model_result', ...
+
+        Specify the upper and lower bounds for certain branch length and rate exchangeability
+        parameter.
+
+        >>> app_alt_params = get_app(
+        ...     "model",
+        ...     "HKY85",
+        ...     tree=tree,
+        ...     param_rules=[
+        ...         {"par_name": "length", "edge": "Human", "upper": 100, "lower": 1e-2},
+        ...         {"par_name": "kappa", "upper": 200, "lower": 1e-6},
+        ...     ],
+        ... )
+        >>> result = app_alt_params(aln2)
+        >>> print(result.lf.get_param_rules()) # doctest: +NORMALIZE_WHITESPACE
+        [{'par_name': 'kappa', ... 'lower': 1e-06, 'upper': 200}, ...
+        {'par_name': 'length', 'edge': 'Human', ... 'lower': 0.01, 'upper': 100}, ...
+
+        Specify the settings in optimiser.
+
+        >>> app_alt_opt = get_app(
+        ...     "model", "HKY85", tree=tree, opt_args=dict(max_restarts=10, tolerance=1e-8)
+        ... )
+        >>> result = app_alt_opt(aln2)
+        >>> print(result.to_rich_dict())
+        {'type': 'cogent3.app.result.model_result', ...
+
+        Specify settings in the likelihood function constructor.
+
+        >>> app_alt_lf = get_app(
+        ...     "model", "HKY85", tree=tree, lf_args = dict(discrete_edges=["Opossum"])
+        ... )
+        >>> result = app_alt_lf(aln2)
+        >>> print(result.to_rich_dict())
+        {... 'likelihood_construction': {... 'discrete_edges': ['Opossum']}, ...
+
+        Split codon and fit models on the relative positions.
+
+        >>> app_sp_codon = get_app(
+        ...     "model", "HKY85", tree=tree, split_codons=True
+        ... )
+        >>> result = app_sp_codon(aln2)
+        >>> print(result.to_rich_dict())
+        {'type': 'cogent3.app.result.model_result', ...
+
+        A NotCompleted object (see https://cogent3.org/doc/app/not-completed.html)
+        is returned if tree(or tree_func) is not provided when seqs number > 3.
+
+        >>> app_notree = get_app("model", "HKY85")
+        >>> result = app_notree(aln2)
+        >>> result.message
+        'to model more than 3, you must provide a tree'
+
+        If the model optimization is unsuccessful, a NotCompleted object will be returned.
```

Comment:
Add the end of this text "(Note that we have deliberately configured the optimiser to raise an exception if it exists because it reached the maximum allowed evaluations.)"